File: packet_seaquence/log/python_shark/src/protocol_analyzer/sctp.py
from src.protocol_analyzer.base import ProtocolAnalyzer

class AnalyzeSctp(ProtocolAnalyzer):
    """
    SCTP内の詳細情報を解析するクラス
    """

    def analyze(self):
        if not hasattr(self.packet, 'sctp'):
            return None
        try:
            sctp_info = {}
            if hasattr(self.packet.sctp, 'srcport'):
                sctp_info['src_port'] = self.packet.sctp.srcport
            if hasattr(self.packet.sctp, 'dstport'):
                sctp_info['dst_port'] = self.packet.sctp.dstport
            if hasattr(self.packet.sctp, 'verification_tag'):
                sctp_info['verification_tag'] = self.packet.sctp.verification_tag
            return sctp_info
        except Exception as e:
            print(f"SCTP解析中にエラーが発生しました: {str(e)}")
            return None

    def get_display_info(self):
        info = self.analyze()
        if not info:
            return "SCTP情報なし"
        return f"SCTP {info.get('src_port','?')} -> {info.get('dst_port','?')} (VTAG={info.get('verification_tag','?')})"

    def get_summary(self):
        info = self.analyze()
        if not info:
            return "SCTP"
        return f"SCTP {info.get('src_port','?')} -> {info.get('dst_port','?')}"

File: packet_seaquence/log/go_src/models/protocol_analyzer/sctp.go
// Package protocol_analyzer provides SCTP protocol analyzer
package protocol_analyzer

import (
	"fmt"
)

// AnalyzeSCTP implements the ProtocolAnalyzer interface for SCTP
type AnalyzeSCTP struct {
	BaseProtocolAnalyzer
}

// NewAnalyzeSCTP creates a new SCTP analyzer
func NewAnalyzeSCTP(packet Packet) *AnalyzeSCTP {
	return &AnalyzeSCTP{
		BaseProtocolAnalyzer: NewBaseProtocolAnalyzer(packet),
	}
}

// Analyze extracts SCTP information from a packet
func (a *AnalyzeSCTP) Analyze() map[string]interface{} {
	if !a.Pkt.LayerExists("sctp") {
		return nil
	}

	sctpInfo := make(map[string]interface{})
	
	// Source port
	if srcPort, ok := a.Pkt.GetField("sctp", "srcport"); ok {
		sctpInfo["src_port"] = srcPort
	}
	
	// Destination port
	if dstPort, ok := a.Pkt.GetField("sctp", "dstport"); ok {
		sctpInfo["dst_port"] = dstPort
	}
	
	// Verification tag
	if verificationTag, ok := a.Pkt.GetField("sctp", "verification_tag"); ok {
		sctpInfo["verification_tag"] = verificationTag
	}
	
	return sctpInfo
}

// GetDisplayInfo returns formatted SCTP information
func (a *AnalyzeSCTP) GetDisplayInfo() string {
	info := a.Analyze()
	if info == nil {
		return "SCTP情報なし"
	}
	
	srcPort := getOrDefault(info["src_port"], "?")
	dstPort := getOrDefault(info["dst_port"], "?")
	verificationTag := getOrDefault(info["verification_tag"], "?")
	
	return fmt.Sprintf("SCTP %s -> %s (VTAG=%s)", srcPort, dstPort, verificationTag)
}

// GetSummary returns a summary of SCTP information
func (a *AnalyzeSCTP) GetSummary() string {
	info := a.Analyze()
	if info == nil {
		return "SCTP"
	}
	
	srcPort := getOrDefault(info["src_port"], "?")
	dstPort := getOrDefault(info["dst_port"], "?")
	
	return fmt.Sprintf("SCTP %s -> %s", srcPort, dstPort)
}